UPGRADE-3.1.md

```diff
@@ -24,6 +24,40 @@ FrameworkBundle
 
  * As it was never an officially supported feature, the support for absolute
    template paths has been deprecated and will be removed in Symfony 4.0.
+ * The core form types registered as services:
```

Comment:
I am still not convinced that it is necessary to add the "use their fully-qualified class name instead" part here (passing the form type instance retrieved from the container already doesn't work).

I think writing "The following services are deprecated since 3.1 and will be removed in 4.0:" is sufficient.